simulation/jsbsim_simulator.py

```diff
@@ -0,0 +1,308 @@
+        restart the simulation with default initial conditions
+    run()
+        run JSBSim at the sim_dt rate
+    get_time()
+        returns the current JSBSim time
+    get_local_position()
+        returns the lat, long and altitude of JSBSim
+    get_local_orientation()
+        returns the euler angle orientation (roll, pitch, yaw) of JSBSim
+    airsim_connect()
+        connect to a running instance of airsim
+    update_airsim()
+        updates the airsim client with the JSBSim calculated pose information
+    close()
+        closes the JSBSim fdm instance
+    start_engines()
+        starts all available aircraft engines
+    set_throttle_mixture_controls()
+        sets aircraft mixture and throttle controls
+    raise_landing_gear()
+        raises the aircraft's landing gear
+    """
+
+    encoding = 'utf-8'
+
+    # Root directory of jsbsim on the computer used for simulation
+    ROOT_DIR = os.path.abspath(r"/Users/owensorber/opt/anaconda3/lib/python3.8/site-packages/jsbsim")
+
+    def __init__(self,
+                 sim_frequency_hz: float = 60.0,
+                 aircraft: Aircraft = x8,
+                 init_conditions: Dict[prp.Property, float] = None,
+                 debug_level: int = 0):
+        self.fdm = jsbsim.FGFDMExec(root_dir=self.ROOT_DIR)
+        self.fdm.set_debug_level(debug_level)
+        self.sim_dt = 1.0 / sim_frequency_hz
+        self.aircraft = aircraft
+        self.initialise(self.sim_dt, self.aircraft.jsbsim_id, init_conditions)
+        self.fdm.disable_output()
+        self.wall_clock_dt = None
+        self.client = self.airsim_connect()
+
+    def __getitem__(self, prop: Union[prp.BoundedProperty, prp.Property]) -> float:
+        return self.fdm[prop.name]
+
+    def __setitem__(self, prop: Union[prp.BoundedProperty, prp.Property], value) -> None:
+        self.fdm[prop.name] = value
+
+    def load_model(self, model_name: str) -> None:
+        """
+        Load a JSBSim xml formatted aircraft model into the JSBSim flight dynamic model
+
+        :param model_name: name of aircraft model loaded into JSBSim
+        :return: None
+        """
+        load_success = self.fdm.load_model(model_name)
+
+        if not load_success:
+            raise RuntimeError('JSBSim could not find specified model name: ' + model_name)
+
+    def get_aircraft(self) -> Aircraft:
+        """
+        Get the Aircraft the JSBSim was initialised with
+
+        :return: aircraft used in the simulator
+        """
+        return self.aircraft
+
+    def get_loaded_model_name(self) -> str:
+        """
+        Get the name of the loaded aircraft model from the current JSBSim FDM instance
+
+        :return: JSBSim model name
+        """
+        name: str = self.fdm.get_model_name().decode(self.encoding)
+        if name:
+            return name
+        else:
+            return None
+
+    def initialise(self, dt: float, model_name: str, init_conditions: Dict['prp.Property', float] = None) -> None:
```

Comment:
Might need to come up with a way to initialize randomly from a reset distribution (not necessarily from a basic takeoff file) as an exploration technique